src/main/java/org/folio/edge/patron/PatronHandler.java

```diff
@@ -259,13 +276,13 @@ protected void handleProxyResponse(RoutingContext ctx, HttpResponse<Buffer> resp
     serverResponse.setStatusCode(statusCode);
 
     String respBody = resp.bodyAsString();
-    if (logger.isDebugEnabled()) {
+    if (logger.isDebugEnabled() ) {
       logger.debug("response: " + respBody);
     }
 
     String contentType = resp.getHeader(HttpHeaders.CONTENT_TYPE.toString());
 
-    if (resp.statusCode() < 400){
+    if (resp.statusCode() < 400 && Objects.nonNull(respBody)){
```

Comment:
what is the use of new condition here?

Reply:
@Vignesh-kalyanasundaram :- just as precaution , There might be a case when we receive an empty or null response from mod-patron. in this case this code will not fail as NullPointer exception .